app/(auth)/signup/_components/signupForm.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import { Controller, FieldValues, useForm } from 'react-hook-form';
+
+import { AxiosError } from 'axios';
+import { useRouter } from 'next/navigation';
+
+import Button from '@/components/button/button';
+import Input from '@/components/input/input';
+import { postSignUpInfo } from '@/services/api';
+
+import MemberType from './memberType';
+
+const EMAIL_REGEX = /^[A-Za-z0-9]([-_.]?[A-Za-z0-9])*@[a-z0-9]([-_.]?[A-Za-z0-9])*\.[A-Za-z]{2,3}$/;
+const PASSWORD_REGEX = /^.{8,}$/;
+
+export default function SignUpForm() {
+  const router = useRouter();
+
+  const {
+    register,
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<FieldValues>({ mode: 'all', defaultValues: { type: 'employee' } });
+
+  const FORM_FIELDS = [
+    {
+      name: 'email',
+      label: '이메일',
+      type: 'text',
+      placeholder: '이메일을 입력해주세요.',
+      validation: {
+        required: '이메일을 입력해주세요.',
+        pattern: {
+          value: EMAIL_REGEX,
+          message: '올바른 이메일 주소가 아닙니다.',
+        },
+      },
+    },
+    {
+      name: 'password',
+      label: '비밀번호',
+      type: 'password',
+      placeholder: '비밀번호를 입력해주세요.',
+      validation: {
+        required: '비밀번호를 입력해주세요.',
+        pattern: {
+          value: PASSWORD_REGEX,
+          message: '비밀번호를 8자 이상 입력해주세요.',
+        },
+      },
+    },
+    {
+      name: 'password_repeat',
+      label: '비밀번호 확인',
+      type: 'password',
+      placeholder: '비밀번호를 한 번 더 입력해주세요.',
+      validation: {
+        validate: (value: string) => value === watch('password') || '비밀번호가 일치하지 않습니다.',
+      },
+    },
+  ];
+
+  const onSubmit = async (data: FieldValues) => {
+    try {
+      await postSignUpInfo(data);
+      router.push('/signin');
+      alert('회원가입이 완료되었습니다.');
+    } catch (error) {
+      const axiosError = error as AxiosError;
+      switch (axiosError.message) {
+        case '400':
+          alert('잘못된 요청입니다. 입력 내용을 확인해주세요.');
+          break;
+        case '409':
+          alert('이미 사용 중인 이메일입니다.');
+          break;
+        default:
+          alert('회원 가입 요청에 실패하였습니다.');
+          break;
+      }
```

Comment:
타입 단언(as AxiosError)으로 처리해주는 것 보다 실제로 error가 어떤 에러인지를 판단하고, 그에 맞게 에러를 다뤄주는게 필요한데요!
따라서 여기서는 아래와 같이 코드를 변경해보면 좋을 것 같아요!
```suggestion
      if (error instanceof AxiosError) {
        switch (error.message) {
          case '400':
            alert('잘못된 요청입니다. 입력 내용을 확인해주세요.');
            break;
          case '409':
            alert('이미 사용 중인 이메일입니다.');
            break;
          default:
            alert('회원 가입 요청에 실패하였습니다.');
            break;
        }
      }
```